### ISSUE TITLE
sporadic failures of the tests within the gorgonnx package

### ISSUE THREAD

**Issue by owulveryck** [owulveryck/onnx-go#21]
this code:

```go
package gorgonnx

import (
        "testing"

        "github.com/owulveryck/onnx-go/backend/testbackend"
        _ "github.com/owulveryck/onnx-go/backend/testbackend/onnx"
)

// TestONNX run the onnx's backend tests against all registered operatos
func TestONNX(t *testing.T) {
        for optype := range operators {
                for _, tc := range testbackend.GetOpTypeTests(optype) {
                        tc := tc // capture range variable
                        t.Run(tc().GetInfo(), tc().RunTest(NewGraph(), true))
                }
        }
}
```

executed with `go test -race` within the [`gorgonnx`](https://github.com/owulveryck/onnx-go/tree/gorgonia/backend/x/gorgonnx) directory of the `gorgonia` branch leads to a race condition ([commit 3bb7f2b](https://github.com/owulveryck/onnx-go/commit/3bb7f2bf454ad0338c49c621994b8d3e18217aec)).

Without the `true` parameter of the `RunTest` method, no race detected, but a weird behavior happens: now and then, one of the tests randomly fails. I suspect this is related to the race condition.

This causes the CI to fail most of the time (which is a good thing though).

**Comment by owulveryck** [comment]
The race condition is within Gorgonia itself; the dimSizerPool is a simple `map` that is not concurrent safe.
See [PR 281](https://github.com/gorgonia/gorgonia/pull/281)

**Comment by owulveryck** [comment]
Found it!

the `iterator.NewOrderedWeightedEdges` do not sort the nodes.
The nodes have to be sorted

**Comment by owulveryck** [comment]
fixed by commit [0b2b3a5aea7ba5d8c46c202b0e1ec9626024c631](https://github.com/owulveryck/onnx-go/pull/20/commits/0b2b3a5aea7ba5d8c46c202b0e1ec9626024c631)
